src/admin/views/Menu/MenuEdit.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, withRouter } from 'react-router';
+
+import { useMutation } from '@apollo/react-hooks';
+import {
+	Button,
+	ButtonToolbar,
+	Flex,
+	Form,
+	FormGroup,
+	IconName,
+	Spinner,
+	TextInput,
+} from '@viaa/avo2-components';
+import { get, startCase } from 'lodash-es';
+
+import { buildLink } from '../../../shared/helpers/generateLink';
+import { ApolloCacheManager } from '../../../shared/services/data-service';
+import { fetchMenuItemById } from '../../../shared/services/menu-service';
+import toastService, { TOAST_TYPE } from '../../../shared/services/toast-service';
+import { AppState } from '../../../store';
+import { ActionsBar } from '../../components';
+import { INSERT_MENU_ITEM, UPDATE_MENU_ITEM_BY_ID } from '../../graphql';
+import { AdminLayout, AdminLayoutActions, AdminLayoutBody } from '../../layouts';
+import { ADMIN_PATH } from '../../routes';
+import { selectMenuItems } from '../../store/selectors';
+import { MenuItem } from '../../types';
+
+interface MenuEditForm {
+	icon: IconName | '';
+	label: string;
+	link: string;
+}
+
+const initialMenuForm = (): MenuEditForm => ({
+	icon: '',
+	label: '',
+	link: '',
+});
+
+interface MenuEditProps extends RouteComponentProps<{ menu: string; id?: string }> {
+	menuItems: MenuItem[];
+}
+
+const MenuEdit: FunctionComponent<MenuEditProps> = ({ history, match, menuItems }) => {
+	const [formErrors, setFormErrors] = useState<Partial<MenuEditForm>>({});
+	const [menuForm, setMenuForm] = useState<MenuEditForm>(initialMenuForm());
+	const [pageType, setPageType] = useState<'edit' | 'create' | undefined>();
+	const [initialMenuItem, setInitialMenuItem] = useState<MenuItem | null>(null);
+	const [isLoading, setIsLoading] = useState<boolean>(true);
+	const [isSaving, setIsSaving] = useState<boolean>(false);
+
+	const [insertMenuItem] = useMutation(INSERT_MENU_ITEM);
```

Comment:
hernoemen naar: triggerMenuItemInsert

Dan blijven we consequent:
![image](https://user-images.githubusercontent.com/1710840/68109909-0ae49400-feec-11e9-92ac-76725227fecb.png)